esp-hal-common/src/dma/mod.rs

```diff
@@ -37,57 +36,34 @@ pub enum DmaPriority {
 
 /// DMA Priorities
 /// The values need to match the TRM
-#[cfg(any(esp32, esp32s2))]
+#[cfg(pdma)]
 #[derive(Clone, Copy)]
 pub enum DmaPriority {
     Priority0 = 0,
 }
 
 /// DMA capable peripherals
 /// The values need to match the TRM
-#[cfg(esp32c2)]
-#[derive(Clone, Copy)]
-pub enum DmaPeripheral {
-    Spi2 = 0,
-    Sha  = 7,
-}
-
-/// DMA capable peripherals
-/// The values need to match the TRM
-#[cfg(esp32c3)]
-#[derive(Clone, Copy)]
-pub enum DmaPeripheral {
-    Spi2  = 0,
-    Uhci0 = 2,
-    I2s   = 3,
-    Aes   = 6,
-    Sha   = 7,
-    Adc   = 8,
-}
-
-/// DMA capable peripherals
-/// The values need to match the TRM
-#[cfg(any(esp32, esp32s2))]
-#[derive(Clone, Copy)]
-pub enum DmaPeripheral {
-    Spi2 = 0,
-    Spi3 = 1,
-}
-
-/// DMA capable peripherals
-/// The values need to match the TRM
-#[cfg(esp32s3)]
 #[derive(Clone, Copy)]
 pub enum DmaPeripheral {
     Spi2   = 0,
+    #[cfg(any(pdma, esp32s3))]
     Spi3   = 1,
+    #[cfg(any(esp32c3, esp32s3))]
     Uhci0  = 2,
+    #[cfg(any(esp32c3, esp32s3))]
     I2s0   = 3,
+    #[cfg(esp32s3)]
     I2s1   = 4,
+    #[cfg(esp32s3)]
     LcdCam = 5,
+    #[cfg(any(esp32c3, esp32s3))]
     Aes    = 6,
+    #[cfg(gdma)]
     Sha    = 7,
+    #[cfg(any(esp32c3, esp32s3))]
     Adc    = 8,
+    #[cfg(esp32s3)]
```

Comment:
not sure this way of defining DmaPeripheral will work fine in future - e.g. on C6 9 is PARLIO while on S3 it's RMT - not much of a problem now but no idea how these IDs will look like for future chips

Reply:
Okay, well at least the existing chips can share a definition I think. If we need to re-introduce a `#[cfg]` and one or more `DmaPeripherals` enums we can do so.